corehq/apps/geospatial/static/geospatial/js/geospatial_map.js

```diff
@@ -358,31 +408,135 @@ hqDefine("geospatial/js/geospatial_map", [
             }
 
             var $saveDrawnArea = $("#btnSaveDrawnArea");
-            $saveDrawnArea.click(function(e) {
+            $saveDrawnArea.click(function () {
                 if (map) {
                     saveGeoJson(map.getMapboxDrawInstance(), mapControlsModelInstance);
                 }
             });
 
             var $exportDrawnArea = $("#btnExportDrawnArea");
-            $exportDrawnArea.click(function(e) {
+            $exportDrawnArea.click(function () {
                 if (map) {
-                    mapControlsModelInstance.exportGeoJson()
+                    mapControlsModelInstance.exportGeoJson();
                 }
             });
-        };
+        }
 
-        var missingGPSModel = function(cases) {
+        var missingGPSModel = function (cases, users) {
             this.casesWithoutGPS = ko.observable(cases);
+            this.usersWithoutGPS = ko.observable(users);
+        };
+        var missingGPSModelInstance = new missingGPSModel([], []);
+
+        var userFiltersModel = function () {
+            var self = {};
+
+            self.shouldShowUsers = ko.observable(false);
+            self.hasFiltersChanged = ko.observable(false);  // Used to disable "Apply" button
+            self.showFilterMenu = ko.observable(true);
+            self.hasErrors = ko.observable(false);
+
+            self.loadUsers = function () {
+                map.removeUserMarkers();
+                self.hasErrors(false);
+                if (!self.shouldShowUsers()) {
+                    self.hasFiltersChanged(false);
+                    return;
+                }
+
+                $.ajax({
+                    method: 'GET',
+                    url: initialPageData.reverse('get_users_with_gps'),
+                    success: function (data) {
+                        self.hasFiltersChanged(false);
+
+                        const usersWithoutGPS = data.user_data.filter(function (item) {
+                            return item.gps_point === null || !item.gps_point.length;
+                        });
+                        missingGPSModelInstance.usersWithoutGPS(usersWithoutGPS);
+
+                        const usersWithGPS = data.user_data.filter(function (item) {
+                            return item.gps_point !== null && item.gps_point.length;
+                        });
```

Comment:
As mentioned in [this](https://github.com/dimagi/commcare-hq/pull/33488#discussion_r1329891353) comment, this functionality was copied over from what we are currently doing for cases. I do agree this could use some refactoring, and I've added a TODO to follow this up in a separate PR. It would then be good to also look at doing a similar refactor for cases.